### PR TITLE
fix: capture and log ci update errors

### LINKED ISSUE #1964: dredd init fails when travis ci integration is requested
**Describe the bug**

Dredd successfully detects Travis CI configuration toward the end of the `dredd init` configuration process. However, if I select `Y` to request integration, `dread init` silently fails to produce a configuration file. If I select `n` everything works as expected.

**To Reproduce**

1. Execute `dredd init` within a Travis CI project,
2. follow the `init` prompts until we reach `Found Travis CI configuration, do you want to add Dredd?`,
3. select `Y` to add Travis CI to Dredd.

**Expected behavior**

A `dredd.yml` configuration file to be created and a log saying "Configuration saved".

**What is in your `dredd.yml`?**

File not created, and no log to indicate configuration is saved.

**What's your `dredd --version` output?**

```
dredd v14.0.0 (Darwin 20.5.0; x64)
```

**Does `dredd --loglevel=debug` uncover something?**

No output is produced after selecting `Y` to `Found Travis CI configuration, do you want to add Dredd?`, regardless of whether debug logging is enabled.

**Can you send us failing test in a Pull Request?**

I may be able to look into this later, but do not have a PR readily available.

### PR DESCRIPTION
**Pull request by rrhodes**
#### :rocket: Why this change?

Errors updating Travis CI configuration files were neither logged nor raised to indicate a problem occurred. Instead, Dredd silently failed without creating a `.dredd.yml` file.

#### :memo: Related issues and Pull Requests

Fixes #1964 

#### :white_check_mark: What didn't I forget?

No unit tests added or modified as part of this change. If reviewers believe this change warrants unit test(s), let me know and I'll be happy to revisit.

- [x] To write docs
- [x] To write tests
- [x] To put [Conventional Changelog](https://dredd.org/en/latest/internals.html#sem-rel) prefixes in front of all my commits and run `npm run lint`.

### CODE REVIEW

**Comment by rrhodes** [comment]
Hey @abtris, I see the most recently merged PR was approved by yourself. Would you mind reviewing this PR for me, please, or pointing me in the right direction for people to reach out to?

**Comment by abtris** [comment]
@rrhodes can you split `packages/dredd/lib/init.js` changes and not necessary docs changes (lowercase docs etc.), please. I don't have issue merge this change but others I see as issue.

**Comment by rrhodes** [comment]
> @rrhodes can you split `packages/dredd/lib/init.js` changes and not necessary docs changes (lowercase docs etc.), please. I don't have issue merge this change but others I see as issue.

Thanks for the prompt feedback, @abtris, appreciate it. I've extracted the documentation changes into new PR #1970 for a separate discussion - this one now only includes the `init.js` change.

**Comment by abtris** [comment]
Please rebase with current master, that should make all tests pass.

**Comment by rrhodes** [comment]
> Please rebase with current master, that should make all tests pass.

Thanks for merging in the fixes to testing, great to see that step going green again! That should be the latest `master` changes applied now.

**Comment by rrhodes** [comment]
Hi @abtris, this PR is ready for your review again - thanks for the feedback so far. The linked docs PR is also awaiting review if either yourself or another maintainer may take a look, please?

**Review by abtris — APPROVED** [review]

**Comment by abtris** [comment]
Waiting for checks to pass, after that I merge it.

**Comment by rrhodes** [comment]
> Waiting for checks to pass, after that I merge it.

Thanks @abtris 🙂

**Pull request merged**
